### PR TITLE
Draft form publishing

### PR DESCRIPTION
**Pull request by l-gorman**
Adding the possibility for live and draft forms to coexist

Main Changes

1. Updated the `Form` Model to include `live` (Boolean), `draftVersion` (String), and `liveVersion` (String). 
2. Adding the ability to count draft and live surveys
3. Adding migrations to update existing surveys in the DB

Minor Change

1. Have removed the ability to automatically publish drafts when uploaded.

### CODE REVIEW

File: routes/forms.js [hunk 1]
```diff
@@ -199,8 +225,25 @@ router.post("/new", auth, async (req, res, next) => {
...
+            formVersion === Number(form.draftVersion) + 1
+
+        }
+
+        else if (form.live==true){
+            formVersion === Number(form.liveVersion) + 1
+
+        }else {
+            throw new HttpError("Could not find a version to assign to this form", 500)
+        }
```
Comment (dave-mills) [inline routes/forms.js:246]: If the form is brand new, there won't be any draft or live version, so can we default to formVersion = 1?

**Review by dave-mills — COMMENTED** [review]
I'm still overall thinking the form versioning is more complicated than it needs to be, but this does give us the benefit of letting us tell the user "Version 5 is currently published. The draft is currently version 8". So if that's the goal here, then cool :)


Overall, I would like to be able to have the auth api handle a 'default' case for form versioning (i.e., new forms have version = 1, then you increment by 1 for each  new draft you save). Later, we might have the option in the survey builder for the user to explicitly set a version, which would override that (by adding the `req.query.form_version` to the `/new` or `/new-draft` request). If/when we add that to the survey builder, I'd make sure there was validation to ensure that the new version number was higher than the existing version(s). 


I definitely like the ability to see the draft and live submission counts 👍

**Comment by l-gorman** [comment]
Thanks @dave-mills, that's really helpful 🙂. I still think it is useful to have the version numbers, it makes it easier for readers to see whether one version is published and whether one is in draft state. I have tried to present this in a clear way for the updated front-end too, (see [here](https://github.com/l-gorman/rhomis-data-app/pull/9)). I have added handling for the default case, it seems to work (when combined with the change to the front end), let me know if works for you.

## Notes

I have created some migration scripts in the "migrations" folder, to make changes to existing records in the DB, you may want to make use of these.

## Known limitations

* Need to add authentication for the "publish" route, will add to issues.

File: routes/forms.js [hunk 2]
```diff
@@ -300,14 +366,15 @@ router.post("/new", auth, async (req, res, next) => {
         const formInformation = {
             name: req.query.form_name,
             project: req.query.project_name,
-            formVersion: formVersion,
+            draftVersion: formVersion,
             users: [req.user._id],
             centralID: centralResponse.data.xmlFormId,
             draft: !publish,
+            live: publish,
```
Comment (dave-mills) [inline routes/forms.js:373]: I think this is the cause of the issue I mentioned in the [data app PR](https://github.com/l-gorman/rhomis-data-app/pull/9). Here, publish needs to be defined. 

Given this is the 'new' endpoint, can we assume that draft should always equal true and live should always equal false? (As there is no chance a user will create a form and publish straight to live without having a draft version first).

Comment (l-gorman) [reply routes/forms.js:373]: Well spotted, yes fully agree, will change now

**Comment by dave-mills** [comment]
Quick update for posterity (so it's here and not just on Slack) - reviewed locally with the modification and was able to go through the whole sequence with the survey builder:
- create a new form and save it as a draft;
- checked the draft appears on the Rhomis app and I could collect data with the QR code;
- finalised the form in the survey builder;
- checked the live form is available for data collection;
- made a change and saved a new draft in the survey buidler;
- checked both live and draft versions are availble for collection;

All working nicely - looks great!

**Comment by dave-mills** [comment]
(But took it out of draft by accident - I'll leave that for you to decide!)

**Pull request merged**